### PR TITLE
fix(radio-button): display validation message in parent group

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #9220

## Summary

Fix validation messages not displaying on `calcite-radio-button-group` during form submission.

Note: `calcite-radio-button`s need to be in a group, or else they will not display validation messages based on the `required` constraint.

The formAssociated common test needs to be updated to support a parent component wrapping the form associated element. Currently it assumes the first element it finds will be the form associated component to test. I will do this in a follow up PR, likely after the release.

### CODE REVIEW

**Comment by driskull** [comment]
Is there an issue for this PR? I think we should have one for verification and tracking purposes.

**Comment by benelan** [comment]
> Is there an issue for this PR? I think we should have one for verification and tracking purposes.

Good point, I created an issue and linked it in the PR body!

File: packages/calcite-components/src/tests/commonTests/formAssociated.ts [hunk 1]
```diff
@@ -436,18 +444,35 @@ export function formAssociated(
...
 
-  async function expectValidationIdle(element: E2EElement) {
-    expect(await element.getProperty("status")).toBe("idle");
-    expect(await element.getProperty("validationMessage")).toBe("");
-    expect(await element.getProperty("validationIcon")).toBe(false);
-  }
+  async function expectValidationProps(page: E2EPage, element: E2EElement, validationProps?: ValidationProps) {
+    let testProps = validationProps;
+
+    if (element.nodeName === "CALCITE-RADIO-BUTTON") {
```
Comment (jcfranco) [inline packages/calcite-components/src/tests/commonTests/formAssociated.ts:457]: Could you also add the explainer comment from `form.tsx`?

**Review by jcfranco — APPROVED** [review]
🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖
🤖✅🤖🤖🤖✅🤖🤖✅✅🤖🤖✅✅✅🤖🤖✅✅✅🤖✅✅✅✅🤖✅🤖
🤖✅✅🤖🤖✅🤖✅🤖🤖✅🤖🤖✅🤖🤖✅🤖🤖🤖🤖✅🤖🤖🤖🤖✅🤖
🤖✅🤖✅🤖✅🤖✅🤖🤖✅🤖🤖✅🤖🤖✅🤖🤖🤖🤖✅✅✅🤖🤖✅🤖
🤖✅🤖🤖✅✅🤖✅🤖🤖✅🤖🤖✅🤖🤖✅🤖🤖🤖🤖✅🤖🤖🤖🤖🤖🤖
🤖✅🤖🤖🤖✅🤖🤖✅✅🤖🤖✅✅✅🤖🤖✅✅✅🤖✅✅✅✅🤖✅🤖
🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖🤖

**Pull request merged**